### PR TITLE
A CLI for running specific tests suites, like bootstrap CLI

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: #1734

## What I did
I improved the 'running-tests'-experience by creating a inquirer script for running the test-suites.

## How to test
1. Install
2. bootstrap
3. run `yarn test` >> should open an inquirer
4. run `yarn test -- --watch` >> should start default suites in watchmode
5. run `yarn test -- --core --watch` >> should start only core suite in watchmode
6. run `yarn test -- --core --reactnative` >> should start core & reactnative suites to completion

It's also possible to run `yarn test -- -h` to get a list of all options available.

## Why
I like this because it makes it discoverable. It also allows contributors to focus on a subset of tests. This is useful in our codebase because changes to for example the `react`, `vue` or `angular` apps can not have any effect on their tests. So if your changes are solely on those parts of the codebase it makes little sense to have to bootstrap the react-native example(s) just to be able to run the unit tests.

I've set ALL the suites as `default = true`, so that includes the react-native suite! So it's an OPT-OUT to NOT run the tests for react-native. Feedback on this would be appreciated very much!

## TODO
- [x] update documentation
- [x] change .circleci.yml

**Is this testable with jest or storyshots?**
Possibly, I'm hoping the reviewers will let me know if they think that would be sensible

**Does this need an update to the documentation?**
YES, I'll do that after I hear positive reactions on the feature itself

Merging this would fix: #1734

### CODE REVIEW

**Review by danielduan — APPROVED** [review]
I tested both cra and react-native. Looks good, really appreciate this!

Can we perhaps have an option to run it with the `-u` to update snapshots?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -22,17 +22,44 @@ No software is bug free. So, if you got an issue, follow these steps:
 
 To test your project against the current latest version of storybook, you can clone the repository and link it with `yarn`. Try following these steps:
 
-1.  Download the latest version of this project, and build it:
+#### 1. Download the latest version of this project, and build it:
+
+```sh
+git clone https://github.com/storybooks/storybook.git
+cd storybook
+yarn install
```
Comment (danielduan) [inline CONTRIBUTING.md]: I think @Hypnosphi opted to use the shorthand `yarn` notation. We should keep it consistent

File: .circleci/config.yml [hunk 2]
```diff
@@ -147,8 +147,8 @@ jobs:
       - run:
           name: "Unit testing"
           command: |
-            yarn test -- --coverage -i
-            yarn coverage
+            npm run test -- --all --coverage --runInBand
```
Comment (danielduan) [inline .circleci/config.yml]: If yarn works on circleci, I think we should keep it consistent there as well.

**Review by danielduan — APPROVED** [review]

**Review by Hypnosphi — CHANGES_REQUESTED** [review]
Same concerns as @danielduan's, otherwise LGTM

**Review by usulpro — APPROVED** [review]
Works fine!
Maybe it worth to add an option for testing a certain package? And do same for test coverage option

**Comment by ndelangen** [comment]
@Hypnosphi should be fixed!

**Comment by Hypnosphi** [comment]
And `yarn` shorthand, please

**Comment by ndelangen** [comment]
I like the explicitness of `yarn install` actually..

**Comment by Hypnosphi** [comment]
Change it to this everywhere then =) Let's be consistent with it

**Review by Hypnosphi — APPROVED** [review]

**Pull request merged**